### PR TITLE
Release 0.7.3

### PR DESCRIPTION
**Pull request by khoroshevskyi**
Changes:
- Fixed POP update
- Improved error handling in views API
- Added stats method to Namespace module
- Updated docs
- Added coverage

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -36,9 +36,6 @@ jobs:
     - name: Install dev dependancies
```
Comment (donaldcampbelljr) [inline .github/workflows/pytest.yml:36]:
```suggestion
    - name: Install dev dependencies
```

File: README.md [hunk 2]
```diff
@@ -1,15 +1,140 @@
...
+
+**Documentation**: <a href="https://pep.databio.org" target="_blank">https://pep.databio.org</a>
+
+**Source Code**: <a href="https://github.com/pepkit/pepdbagent" target="_blank">https://github.com/pepkit/pepdbagent</a>
+
+---
 
 `pepdbagent` is a Python library and toolkit that gives a user-friendly 
-interface to connect, upload, update and retrieve information from pep-db.
+interface to connect, upload, update and retrieve information from pep database. This library is designed to work
```
Comment (donaldcampbelljr) [inline README.md:22]:
```suggestion
interface to connect, upload, update and retrieve information from the pep database. This library is designed to work 
```

File: README.md [hunk 3]
```diff
@@ -1,15 +1,140 @@
...
+**Documentation**: <a href="https://pep.databio.org" target="_blank">https://pep.databio.org</a>
+
+**Source Code**: <a href="https://github.com/pepkit/pepdbagent" target="_blank">https://github.com/pepkit/pepdbagent</a>
+
+---
 
 `pepdbagent` is a Python library and toolkit that gives a user-friendly 
-interface to connect, upload, update and retrieve information from pep-db.
+interface to connect, upload, update and retrieve information from pep database. This library is designed to work 
+to be used by PEPhub, but it can be used for any other purpose, to manage data in pep database.
```
Comment (donaldcampbelljr) [inline README.md:23]:
```suggestion
with PEPhub, but it can be used for any other purpose.
```

File: README.md [hunk 4]
```diff
@@ -1,15 +1,140 @@
...
+
+pepdbagent creates a connection to the database and creates table schemas for the PEPhub database if necessary.
```
Comment (donaldcampbelljr) [inline README.md:25]:
```suggestion
`pepdbagent` creates a connection to the database and creates table schemas for the PEPhub database if necessary.
```

File: README.md [hunk 5]
```diff
@@ -1,15 +1,140 @@
...
+Core database is `postgres` database, but it can be easily extended to other relational databases.
+To use `pepdbagent`, you need to have a database instance running with it's credentials.
```
Comment (donaldcampbelljr) [inline README.md:27]:
```suggestion
To use `pepdbagent`, you need to have a database instance running with its credentials.
```

File: README.md [hunk 6]
```diff
@@ -1,15 +1,140 @@
...
+Example:
+```python
+# Get info about namespace by providing query argument. Then pepdbagent will
+# search for a specified pattern of namespace in database.
+agent.namespace.get(query='Namespace')
 
-Then, follow the `pepdbagent` tutorial here:
-- [pedbagent](./docs/README.md)
\ No newline at end of file
+# By default all get function will retrun namespace information for public projects,
```
Comment (donaldcampbelljr) [inline README.md:131]:
```suggestion
# By default all get functions will return namespace information for public projects,
```

File: pepdbagent/modules/namespace.py [hunk 7]
```diff
@@ -204,3 +212,54 @@ def info(self, limit: int = DEFAULT_LIMIT_INFO) -> ListOfNamespaceInfo:
             limit=limit,
             results=list_of_results,
         )
+
+    def stats(self, namespace: str = None, monthly: bool = False) -> NamespaceStats:
+        """
+        Get statistics for project in the namespace or for all projects in the database.
+
+        :param namespace: namespace name [Default: None (all projects)]
+        :param monthly: if True, get statistics for the last 3 years monthly, else for the last 3 months daily.
```
Comment (donaldcampbelljr) [inline pepdbagent/modules/namespace.py:221]: Just curious, why these default values? 3 years monthly vs 3 months daily.
Why not just the last year? or the last 6 months if monthly?

File: pepdbagent/modules/namespace.py [hunk 8]
```diff
@@ -204,3 +212,54 @@ def info(self, limit: int = DEFAULT_LIMIT_INFO) -> ListOfNamespaceInfo:
...
+        """
+        if monthly:
+            number_of_month = 3
+        else:
+            number_of_month = 12 * 3
+        today_date = datetime.today().date() + timedelta(days=1)
+        three_month_ago = today_date - timedelta(days=number_of_month * 30 + 1)
+        statement_update = select(Projects.last_update_date).filter(
```
Comment (donaldcampbelljr) [inline pepdbagent/modules/namespace.py:229]: rename to `statement_last_updated` and `statement_created_date` to be more explicit?

File: pepdbagent/modules/view.py [hunk 9]
```diff
@@ -265,34 +265,32 @@ def add_sample(
...
+                if not view:
+                    raise ViewNotFoundError(
+                        f"View {view_name} of the project {namespace}/{name}:{tag} does not exist"
                     )
-                )
-                sample = sa_session.scalar(sample_statement)
-                if not sample:
-                    raise SampleNotFoundError(
-                        f"Sample {namespace}/{name}:{tag}:{sample_name} does not exist"
+                for sample_name_one in sample_name:
```
Comment (donaldcampbelljr) [inline pepdbagent/modules/view.py:275]: ` for sample_name in sample_names:` would read better

**Review by donaldcampbelljr — APPROVED** [review]
Minor suggestions.

File: pepdbagent/modules/namespace.py (hunk 7 above)
Comment (khoroshevskyi) [reply pepdbagent/modules/namespace.py:221]: idk, I just decided to do it. We can change it in the future. This functionality will be disabled on pephub anyway.
About 3 month, Nathan suggested it

**Pull request merged**